### PR TITLE
[FLINK-30692][sql-client] Introduce SingleSessionManager to support REMOVE JAR syntax

### PR DESCRIPTION
**Pull request by fsk119**
## What is the purpose of the change

*Introduce a specical `SessionManager` for the sql-client embedded mode, which require less resources and add Handler for REMOVE JAR statement.*

## Verifying this change

This change is already covered by existing tests, such as *CliClientITCase*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/SingleSessionManager.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    @Override
+    public Session getSession(SessionHandle sessionHandle) throws SqlGatewayException {
+        Preconditions.checkArgument(
+                session != null && sessionHandle.equals(session.getSessionHandle()),
+                "The specified session doesn't exists");
+        return session;
+    }
+
+    @Override
+    public synchronized Session openSession(SessionEnvironment environment)
```
Comment (LadyForest) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/SingleSessionManager.java:82]: No need to add `synchronized` here?

**Review by LadyForest — CHANGES_REQUESTED** [review]
Thanks for the contribution! And I left some comments.

File: flink-table/flink-sql-client/src/test/resources/sql/set.q [hunk 2]
```diff
@@ -118,14 +118,13 @@ SELECT hive_add_one(1);
 Received a total of 1 row
 !ok
 
-# TODO: support this in the FLINK-30692
-#REMOVE JAR '$VAR_UDF_JAR_PATH';
-#[INFO] The specified jar is removed from session classloader.
-#!info
+REMOVE JAR '$VAR_UDF_JAR_PATH';
```
Comment (LadyForest) [inline flink-table/flink-sql-client/src/test/resources/sql/set.q:121]: Nit: add `SHOW JARS` before and after `REMOVE JAR`?

File: flink-table/flink-sql-client/src/test/resources/sql/set.q [hunk 3]
```diff
@@ -313,3 +303,11 @@ SELECT id, func1(str) FROM (VALUES (1, 'Hello World')) AS T(id, str) ;
 +----+-------------+--------------------------------+
 Received a total of 1 row
 !ok
+
+REMOVE JAR '$VAR_UDF_JAR_PATH';
```
Comment (LadyForest) [inline flink-table/flink-sql-client/src/test/resources/sql/set.q:307]: ditto

File: flink-table/flink-sql-client/src/test/resources/sql/function.q [hunk 4]
```diff
@@ -218,10 +218,9 @@ org.apache.flink.table.api.ValidationException: Alter temporary catalog function
 # test create function using jar
 # ==========================================================================
 
-# TODO: support this in the FLINK-30692
-# REMOVE JAR '$VAR_UDF_JAR_PATH';
-# [INFO] The specified jar is removed from session classloader.
-# !info
+REMOVE JAR '$VAR_UDF_JAR_PATH';
```
Comment (LadyForest) [inline flink-table/flink-sql-client/src/test/resources/sql/function.q:221]: Nit: add `SHOW JARS` before and after `REMOVE JAR`?

File: flink-table/flink-sql-client/src/test/resources/sql/function.q [hunk 5]
```diff
@@ -280,3 +270,11 @@ show user functions;
 +---------------+
 1 row in set
 !ok
+
+REMOVE JAR '$VAR_UDF_JAR_PATH';
```
Comment (LadyForest) [inline flink-table/flink-sql-client/src/test/resources/sql/function.q:274]: ditto

Comment (fsk119) [reply flink-table/flink-sql-client/src/test/resources/sql/function.q:274]: The `show jars` is in the line 244

File: flink-table/flink-sql-client/src/test/resources/sql/set.q (hunk 2 above)
Comment (fsk119) [reply flink-table/flink-sql-client/src/test/resources/sql/set.q:121]: The  show jars statement before the remove jar is at line line 281

File: flink-table/flink-sql-client/src/test/resources/sql/set.q (hunk 3 above)
Comment (fsk119) [reply flink-table/flink-sql-client/src/test/resources/sql/set.q:307]: The show jars statement before the remove jar is at line line 281

**Review by LadyForest — APPROVED** [review]
Thanks for the update, LGTM

**Pull request merged**